src/figuresClasses.ts

```diff
@@ -1,11 +1,86 @@
-export interface Figure {}
+type ValidShapes = 'triangle' | 'circle' | 'rectangle';
+type ValidColors = 'red' | 'green' | 'blue';
 
-export class Triangle implements Figure {}
+export interface Figure {
+  shape: ValidShapes;
+  color: ValidColors;
+  getArea: () => number;
+}
+
+export class Triangle implements Figure {
+  public shape: ValidShapes = 'triangle';
+
+  constructor(
+    public color: ValidColors,
+    public a: number,
+    public b: number,
+    public c: number,
+  ) {
+    const sides = [a, b, c];
+    const longestSide = Math.max(...sides);
+    let sumOfOtherSides = 0;
+
+    for (const side of sides) {
+      if (side < longestSide) {
+        sumOfOtherSides += side;
+      }
+    }
+
+    if (a <= 0 || b <= 0 || c <= 0 || sumOfOtherSides <= longestSide) {
+      throw new Error('is not valid length!');
+    }
+  }
+
+  getArea(): number {
+    const p = (this.a + this.b + this.c) / 2;
+    const triangleArea = Math.sqrt(
+      p * (p - this.a) * (p - this.b) * (p - this.c),
+    );
+
+    return Math.floor(triangleArea * 100) / 100;
+  }
+}
 
-export class Circle implements Figure {}
+export class Circle implements Figure {
+  public shape: ValidShapes = 'circle';
 
-export class Rectangle implements Figure {}
+  constructor(
+    public color: ValidColors,
+    public radius: number,
+  ) {
+    if (radius <= 0) {
+      throw new Error('is not valid length!');
```

Comment:
[CODE STYLE] - When throwing an error, provide a more detailed description in the error message to make it clear what the issue is. For example, 'Circle radius must be a positive number.'